firedrake/matrix_free/operators.py

```diff
@@ -244,24 +318,44 @@ def createSubMatrix(self, mat, row_is, col_is, target=None):
         row_bcs = []
         col_bcs = []
 
-        for bc in self.row_bcs:
-            for i, r in enumerate(row_inds):
-                if bc.function_space().index == r:
-                    row_bcs.append(DirichletBC(Wrow.split()[i],
-                                               bc.function_arg,
-                                               bc.sub_domain,
-                                               method=bc.method))
-
-        if Wrow == Wcol and row_inds == col_inds and self.row_bcs == self.col_bcs:
+        def rcollect_bcs(bc, inds, row_inds, col_inds, Ws):
```

Comment:
I guess you're using `r` to mean "recursive". But that's an implementation detail I think. Why not just call this `collect_bcs`? Or perhaps `collect_sub_bcs`?

Reply:
Fixed.